### DISCUSSION TITLE
Spring Modulith Example

### DISCUSSION THREAD

**Discussion by tschuehly** [microstream-one/microstream#667] [General]
My company currently looks into using MicroStream in combination with [Spring Modulith](https://spring.io/projects/spring-modulith)  to create Highly performant full stack web applications.
I'm interested the best way to structure the application and the object graph in a modular monolith where you have seperated application modules.

**Comment by hg-ms** [comment]
Hello,
As I am not used to Spring Modulith I can’t give you specific tips, just one general requirement:
Microstream needs to access all classes to be peristed with deep reflections. Therefore, every module that contains classes that should be persisted by Microstream must grant the necessary permissions to Microstream.
